Userland/Utilities/network-settings.cpp

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (c) 2023, Fabian Dellwing <fabian@dellwing.net>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <AK/JsonObject.h>
+#include <AK/JsonParser.h>
+#include <LibCore/ConfigFile.h>
+#include <LibCore/System.h>
+#include <LibMain/Main.h>
+
+ErrorOr<int> serenity_main(Main::Arguments)
+{
+    TRY(Core::System::pledge("stdio rpath wpath cpath recvfd sendfd proc exec"));
+
+    TRY(Core::System::unveil("/bin/NetworkServer", "x"));
+    TRY(Core::System::unveil("/etc/Network.ini", "rwc"));
+    TRY(Core::System::unveil(nullptr, nullptr));
+
+    auto infile = TRY(Core::File::standard_input());
+
+    auto input_bytes = TRY(infile->read_until_eof());
+    auto json_data = static_cast<StringView>(input_bytes);
+
+    if (json_data.is_empty() || json_data.is_whitespace())
+        return Error::from_errno(EINVAL);
+
+    auto json = TRY(JsonParser(json_data).parse());
+
+    if (!json.is_object())
+        return Error::from_errno(EINVAL);
+
+    auto json_object = json.as_object();
+
+    auto config_file = TRY(Core::ConfigFile::open_for_system("Network", Core::ConfigFile::AllowWriting::Yes));
+    json_object.for_each_member([&](DeprecatedString const& adapter_name, JsonValue const& adapter_data) {
+        adapter_data.as_object().for_each_member([&](const DeprecatedString& key, const JsonValue& value) {
+            switch (value.type()) {
+            case JsonValue::Type::String:
+                config_file->write_entry(adapter_name, key, value.as_string());
+                break;
+            case JsonValue::Type::Bool:
+                config_file->write_bool_entry(adapter_name, key, value.as_bool());
+                break;
+            case JsonValue::Type::Null:
+                break;
+            default:
+                dbgln("Extend switch/case key={}, value={}", key, value);
+                VERIFY_NOT_REACHED();
+            }
+        });
+    });
+    TRY(config_file->sync());
+
+    auto command = Vector<StringView>();
+    TRY(command.try_append("/bin/NetworkServer"sv));
+    TRY(Core::System::exec_command(command, true));
```

Comment:
Why do we try to run the NetworkServer app from this suid root helper? Isn't that supposed to be started by SystemServer on boot? Looking at NetworkServer/main.cpp, doing this will actually kill any running DHCP client processes (in a kludgy way) and re-launch them? 

eh... I guess that's correct but the design feels like it's missing something. This should probably be sending a message to a "server", not running a configurator tool that messes with the network settings and then exits promptly. Maybe NetworkServer is misnamed? Or maybe we should be updating the existing "Server" application rather than creating a second hop of command line tools to talk to?

... this should probably be a new GitHub issue rather than a blocker on the PR though.

Reply:
This so called "server" is not really a server I think. I just moved the call to it into the new binary, so no change there atm.